### DISCUSSION TITLE
Pocketbase suitable for this project?

### DISCUSSION THREAD

**Discussion by thinkvisual** [pocketbase/pocketbase#1290] [Q&A] [answered]
Thank you for creating Pocketbase! I am really enjoying learning how to use it. 

Would it be possible to use PB to:

a) Integrate with Stripe (to allow different levels of access)
b) Allow real time chat between users 

Or better to use another platform for this?

Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
As always, this depends on your use case and requirements (eg. how many concurrent users do you expect, how large is your dataset, whether vertical-only scaling is ok for your app, etc.).
It should be fine for most small and midsize applications (it can handle several thousands of concurrent reads and writes even on low powered 2GB vps).

Both a) and b) are doable with PocketBase.

a) can be achieved either by [extending PocketBase with Go](https://pocketbase.io/docs/use-as-framework/) OR if you are using SSR you can handle the Stripe webhook in your node server and then update your PocketBase user data. If you haven't already, you could also check https://pocketbase.io/docs/collections/#auth-collection for examples with the possible access controls in PocketBase.

b) can be achieved directly with the [realtime subscriptions api](https://pocketbase.io/docs/api-realtime/).

**With that said, please keep in mind that PocketBase is still in very active development stage and full backward compatibility is not guaranteed before [v1.0.0](https://github.com/orgs/pocketbase/projects/2).**

**Reply by thinkvisual** [reply]
Thank you for such a detailed reply! Appreciate it

**Reply by BobSnrub** [reply]
Referring to answer a), could this be done with Sveltekit?

**Reply by ganigeorgiev** [reply]
@Burns316 Yes, if you use SSR (aka. has a node server-side endpoint that will take care for verifying the Stripe webhook).

**Reply by AndrewDownsie22** [reply]
refering to A again. Using nextjs could I use the built in API route to handle this if I'm not using express

**Reply by ganigeorgiev** [reply]
@AndrewDownsie22 Yes, if you are using a nextjs api route (aka. server endpoint) it should work (for Stripe webhook verification you can check their official example in https://github.com/stripe/stripe-node/blob/master/examples/webhook-signing/node-express/express.js; it uses plain express but the flow should be similar, eg. replace the req.body and req.headers with the nextjs api route request body and headers)

**Reply by AndrewDownsie22** [reply]
@ganigeorgiev Thank you! Excited to start using pocketbase.

**Comment by LoreMontero** [comment]
I'll be hosting a project management website for my motorsport team. I'm expecting to have around 10-30 monthly active users. I'm wondering if pocketbase would be a great option for me. And if it is, I know pocketbase doesn't offer services to host your database but is there an alternative way I could host it with different providers?

**Reply by ganigeorgiev** [reply]
I think it should be fine. PocketBase can handle thousand of concurrent users even on a low powered vps (proper benchmarks will be available soon).

> is there an alternative way I could host it with different providers?

Any hosting provider that offers persistent disk storage would work - [Hetzner](https://www.hetzner.com/), [Vultr](https://www.vultr.com/), [UpCloud](https://upcloud.com/), [Linode](https://www.linode.com/), [DigitalOcean](https://www.digitalocean.com/), etc.

There are also providers that have free tiers like [fly.io](https://github.com/pocketbase/pocketbase/discussions/537), [mogenius.com](https://mogenius.com/), [oracle-cloud](https://www.oracle.com/cloud/free/) and [pockethost.io](https://pockethost.io/) (_I haven't explored in details the last 3 options yet_).
